website/blog/2023-01-20-typescript-first.md

```diff
@@ -0,0 +1,101 @@
+
+With TS types co-located with React Native source, there is more visibility and ownership of TS types. Our team is actively working on tools to maintain alignment between Flow and TS.
+
+This change also removes a dependency for React Native users to manage. When upgrading to 0.71 or above, you can remove `@types/react-native` as a dependency. [Refer to the new app template on how to set up TypeScript support.](https://github.com/facebook/react-native/blob/main/template/tsconfig.json)
+
+We plan to deprecate `@types/react-native` for versions 0.73 and onward. Concretely this means:
+
+- `@types/react-native` tracking React Native versions 0.71 and 0.72 will be released. They will be identical to the types in React Native on the relevant release branches.
+- For React Native 0.73 and onward, TS types will only be available from React Native.
+
+### How to Migrate
+
+Please migrate to the new co-located types at your earliest convenience. Here are more details on migrating based on your needs.
+
+#### App Maintainer
+
+Once you upgrade to React Native >= 0.71, you can remove the `@types/react-native` from your `devDependency`.
+
+:::note
+
+If you have warnings because a library you use references `@types/react-native` as a `peerDependency`, file an issue or open a PR for that library to use [optional peerDependencies](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#peerdependenciesmeta) and ignore the warning for now.
+
+:::
+
+#### Library Maintainer
+
+Libraries that target versions of React Native below 0.71 may use a `peerDependency` of `@types/react-native` to typecheck against the apps version of typings. This dependency should be marked as optional in [`peerDependenciesMeta`](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#peerdependenciesmeta) so that the typings are not required for users without TypeScript or for 0.71 users where typings are built-in.
+
+#### Maintainer of TypeScript declarations that depend on `@types/react-native`
+
+Check out the [breaking changes introduced with 0.71](https://github.com/facebook/react-native/blob/main/CHANGELOG.md) to see if you’re ready to migrate.
+
+### What if I use Flow?
+
+Flow users can continue to typecheck applications targeting 0.71+ but configuration logic for it is no longer included out-of-the box in the template.
+
+Flow users have previously upgraded React Native’s Flow types by merging in the `.flowconfig` from the new app template and manually updating `flow-bin`. The new app template no longer has a `.flowconfig`, but [one is still packaged within React Native](https://github.com/facebook/react-native/blob/main/.flowconfig) that can be used as a basis for your app.
```

Comment:
I went to double check this, and the `.flowconfig` is not included as part of the NPM packages `files` field, so it is part of the repo but not part of the NPM package. We could change this if we wanted, but otherwise could probably update wording to specify that its part of the repo but not the package.

```suggestion
Flow users have previously upgraded React Native’s Flow types by merging in the `.flowconfig` from the new app template and manually updating `flow-bin`. The new app template no longer has a `.flowconfig`, but [one is still present in the React Native repository](https://github.com/facebook/react-native/blob/main/.flowconfig) that can be used as a basis for your app.
```